com.unity.cinemachine/Editor/Upgrader/CinemachineUpgrader.cs

```diff
@@ -618,60 +782,135 @@ static void ConvertCinemachinePathToSpline(CinemachinePathBase pathBase, out Spl
+                    
+                                var prefabInstance = component.gameObject;
+                                var r1 = PrefabUtility.GetCorrespondingObjectFromSource(prefabInstance);
+                                if (x.Equals(r1))
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                        return false;
+                    }
                 });
-
-                prefabCount = m_PrefabVcams.Count;
+                
+                prefabCount = m_PrefabAssets.Count;
                     
 #if DEBUG_HELPERS
                 Debug.Log("**** All prefabs ****");
-                m_PrefabVcams.ForEach(prefab => Debug.Log(prefab.name));
+                m_PrefabRoots.ForEach(prefab => Debug.Log(prefab.name));
                 Debug.Log("*********************");
 #endif
             }
+
+            public static List<GameObject> FindAllInstancesOfPrefabEvenInNestedPrefabs(Scene activeScene, GameObject prefab)
+            {
+                // find all freelooks and vcams in the current scene
+                var rootObjects = activeScene.GetRootGameObjects();
+                var componentsList = new List<CinemachineVirtualCameraBase>();
+                foreach (var go in rootObjects)
+                {
+                    componentsList.AddRange(go.GetComponentsInChildren<CinemachineFreeLook>(true).ToList());
+                }
+
+                foreach (var go in rootObjects)
+                {
+                    componentsList.AddRange(go.GetComponentsInChildren<CinemachineVirtualCamera>(true).ToList());
+                }
+
+                var allInstances = new List<GameObject>();
+                foreach (var component in componentsList)
+                {
+                    if (component.ParentCamera is CinemachineFreeLook)
+                        continue; // ignore, because it is a freelook rig
+                    
+                    var prefabInstance = component.gameObject;
+                    if (prefabInstance.name.Contains(k_UnupgradableTag))
+                        continue; // ignore, because we created it
+                    
+                    var source = PrefabUtility.GetCorrespondingObjectFromSource(prefabInstance);
+                    // Check if prefabInstance is an instance of the input prefab
+                    // If prefabInstance is a direct instance of prefab, then source is the same object as prefab
+                    // However, if prefabInstance is part of a nested prefab, then source is unfortunately a different object.
+                    // So I have to compare their names, which should be a sufficient condition, it does not make sense
+                    // to have prefabs assets with the same name, but it is not enforced, so it could happen...
+                    if (prefab.Equals(source) || prefab.name == source.name) 
```

Comment:
I'm getting a null exception here.  It seems that GetCorrespondingObjectFromSource can return null

Reply:
Yes, that was fixed in the gigaya PR